### PR TITLE
Added Purpose and logic for FisAccount.

### PR DESCRIPTION
**Pull request by taylorkj**
@srkirkland 
Extra fields and logic necessary to reduce the number of left outer joins to build final object.

### CODE REVIEW

File: GiftModels/FinancialDetails.cs [hunk 1]
```diff
@@ -32,7 +32,28 @@ public class FinancialDetails
...
         public virtual string FundType { get; set; }
 
-        public virtual string FisAccount { get; set; }
+        public virtual string Purpose { get; set; }
+
+        public virtual string FisAccount {
+            get
+            {
+                string retval = null;
+                if (FundType.Equals("CURRENT") && !(Purpose.Equals("LOAN") || Purpose.Equals("CAPITAL")))
```
Comment (srkirkland) [inline GiftModels/FinancialDetails.cs]: @taylorkj you should check to make sure fundType or Purpose is not null, or else it'll throw an exception.  Alternately you could use `string.Equals(FundType, "CURRENT")`

File: GiftModels/FinancialDetails.cs [hunk 2]
```diff
@@ -32,7 +32,28 @@ public class FinancialDetails
...
         public virtual string FundType { get; set; }
 
-        public virtual string FisAccount { get; set; }
+        public virtual string Purpose { get; set; }
+
+        public virtual string FisAccount {
+            get
+            {
+                string retval = null;
+                if (string.Equals(FundType, "CURRENT") && !(string.Equals(Purpose, "LOAN") || string.Equals(Purpose, "CAPITAL")))
```
Comment (jSylvestre) [inline GiftModels/FinancialDetails.cs]: @taylorkj Case insensitive?

**Comment by srkirkland** [comment]
@taylorkj :ship:

File: GiftModels/FinancialDetails.cs [hunk 3]
```diff
@@ -32,7 +32,28 @@ public class FinancialDetails
 
         public virtual string FundType { get; set; }
 
-        public virtual string FisAccount { get; set; }
+        public virtual string Purpose { get; set; }
+
+        public virtual string FisAccount {
```
Comment (jSylvestre) [inline GiftModels/FinancialDetails.cs]: @jpknoll When I look at the meta data for this in Giving, all I see is a get;set; Not this code. Is it actually being called?

**Pull request merged**